src/Workspaces/Remote/ServiceHub/Services/ExtensionMethodImportCompletion/RemoteExtensionMethodImportCompletionService.cs

```diff
@@ -40,9 +38,13 @@ public RemoteExtensionMethodImportCompletionService(in ServiceConstructionArgume
             bool hideAdvancedMembers,
             CancellationToken cancellationToken)
         {
+            var stopwatch = SharedStopwatch.StartNew();
             return RunServiceAsync(solutionChecksum, async solution =>
             {
-                var document = solution.GetDocument(documentId)!;
+                var assetSyncTime = stopwatch.Elapsed;
+
+                // Completion always uses frozen-partial semantic in-proc, which is not automatically passed to OOP, so enable it explicitly
+                var document = solution.GetRequiredDocument(documentId).WithFrozenPartialSemantics(cancellationToken);
```

Comment:
As @CyrusNajmabadi mentioned above, this is a no-op, but I'd like to keep it just to make it clear that this is a place we could use partial semantic to avoid delays.